internal/dump/zip.go

```diff
@@ -3,36 +3,30 @@ package dump
 import (
 	"archive/zip"
 	"context"
-	"io"
 	"path/filepath"
 
-	"github.com/restic/restic/internal/bloblru"
 	"github.com/restic/restic/internal/errors"
 	"github.com/restic/restic/internal/restic"
 )
 
-type zipDumper struct {
-	cache *bloblru.Cache
-	w     *zip.Writer
-}
+func (d *Dumper) dumpZip(ctx context.Context, ch <-chan *restic.Node) (err error) {
+	w := zip.NewWriter(d.w)
 
-// Statically ensure that zipDumper implements dumper.
-var _ dumper = &zipDumper{}
+	defer func() {
+		if err == nil {
+			err = w.Close()
```

Comment:
Shouldn't this error also be wrapped just like in `dumpTar`?

Reply:
Done.